ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -470,7 +470,12 @@ impl<'a> TraceBuilder<'a> {
                 args.push(self.handle_operand(arg)?);
             }
             let jit_func_decl_idx = self.handle_func(inst.callee())?;
-            let instr = if !inst.callee().func_type(self.aot_mod).is_vararg() {
+            let instr = if !self
```

Comment:
Again, this makes the code much harder to read at a glance.

The helper serves to lift away the complexity from the call-site, making it easier to comprehend.

Reply:
[I confused this one earlier.]

This helper *really* confused me, because it's the only index type that can return a non-index thing. It doesn't help that we have `Func`, `FuncIdx`, and `FuncType` in the mix too.

Reply:
>  because it's the only index type that can return a non-index thing

It just makes the IR easier to work with, don't you think? It means we don't have to explicitly convert indices to what they refer to all the time.

Reply:
This is the only helper on an index and I spent ages baffled so "no" vote from me!

Reply:
There are many similar helpers on index types in the JIT IR.

We need to move on. What shall we do:
 - Remove all helpers and inline them at all call-sites
 - Keep the helpers.

Reply:
This helper is bad because it's on an `Idx` type. The other helpers are more a matter of taste.